### ISSUE TITLE
[mtt] hang in osu_micro_benchmark_collective/osu_iallgather with TLS=rc_x

### ISSUE THREAD

**Issue by amaslenn** [openucx/ucx#2469]
Configuration:
```
OMPI: 3.1.0rc4
hercules x8 (ppn=32(x8), nodelist=clx-hercules-[001-008])
Job: ucx-hwtm-rc
```

MTT log: http://e2e-gw.mellanox.com:4080//mnt/lustre/users/mtt/scratch/ucx_ompi/20180401_040409_15533_71654_clx-hercules-001/html/test_stdout_hh3Cpz.txt

Cmd:
`mpirun -np 32 -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' --bind-to core -x UCX_NET_DEVICES=mlx5_2:1 -x UCX_IB_REG_METHODS=rcache,direct -x UCX_TLS=rc_x,sm -x UCX_RC_MLX5_TM_ENABLE=y -mca pmix_base_async_modex 1 -mca mpi_add_procs_cutoff 0 -mca pmix_base_collect_data 0 --map-by node /mnt/lustre/users/mtt/scratch/ucx_ompi/20180401_040409_15533_71654_clx-hercules-001/installs/CHiB/tests/osu_micro_benchmark/osu-micro-benchmarks-5.3.2/mpi/collective/osu_iallgather`

Output:
```
Sun Apr  1 10:55:33 2018[1,0]<stdout>:
Sun Apr  1 10:55:33 2018[1,0]<stdout>:# OSU MPI Non-blocking Allgather Latency Test v5.3.2
Sun Apr  1 10:55:33 2018[1,0]<stdout>:# Overall = Coll. Init + Compute + MPI_Test + MPI_Wait
Sun Apr  1 10:55:33 2018[1,0]<stdout>:
Sun Apr  1 10:55:33 2018[1,0]<stdout>:# Size           Overall(us)       Compute(us)    Pure Comm.(us)        Overlap(%)
Sun Apr  1 10:55:33 2018[1,0]<stdout>:1                      35.11             18.13             16.88              0.00
Sun Apr  1 10:55:33 2018[1,0]<stdout>:2                      34.97             17.99             16.82              0.00
Sun Apr  1 10:55:33 2018[1,0]<stdout>:4                      35.07             18.00             16.83              0.00
Sun Apr  1 10:55:33 2018[1,0]<stdout>:8                      35.15             18.01             16.86              0.00
Sun Apr  1 10:55:34 2018[1,0]<stdout>:16                     35.26             18.12             16.96              0.00
Sun Apr  1 10:55:34 2018[1,0]<stdout>:32                     36.57             18.78             17.54              0.00
Sun Apr  1 10:55:34 2018[1,0]<stdout>:64                     43.59             22.40             21.06              0.00
Sun Apr  1 10:55:34 2018[1,0]<stdout>:128                    46.64             23.96             22.53              0.00
Sun Apr  1 10:55:34 2018[1,0]<stdout>:256                    50.97             25.96             24.47              0.00
Sun Apr  1 10:55:34 2018[1,0]<stdout>:512                    53.74             27.94             26.38              2.23
```

Works well with `NP=31`. Also works with `yalla`, and all TLS except `rc_x`: `rc`, `ud`, `dc` and `dc_x`.

**Comment by yosefe** [comment]
also reproduced by
`mpirun -np 32 -mca btl self -mca pml ucx -mca coll '^hcoll' -x UCX_NET_DEVICES=mlx5_2:1 -x UCX_TLS=rc_x,sm -mca pmix_base_async_modex 1 -mca mpi_add_procs_cutoff 0 -mca pmix_base_collect_data 0 --map-by node /mnt/lustre/users/mtt/scratch/ucx_ompi/20180409_113310_22596_74493_clx-hercules-001/installs/SR4R/tests/osu_micro_benchmark/osu-micro-benchmarks-5.3.2/mpi/collective/osu_iallgather`

looks related to direct modex and ud-based wireup

**Comment by yosefe** [comment]
fixed by #2519
